packages/alfa-html/src/alphabet.ts

```diff
@@ -237,6 +264,522 @@ const tagName: Pattern = (stream, emit, state) => {
+  }
+
+  function anythingElse() {
+    emit({ type: TokenType.Character, data: "<" });
+    emit({ type: TokenType.Character, data: "/" });
+    for (let i = 0; i < state.temporaryBuffer.length; i++) {
+      emit({ type: TokenType.Character, data: state.temporaryBuffer[i] });
+    }
+    return scriptDataEscaped;
+  }
+  const char = stream.peek(0);
+  if (char === null) {
+    return anythingElse;
+  }
+  switch (char) {
+    case Char.CharacterTabulation:
+    case Char.LineFeed:
+    case Char.FormFeed:
+    case Char.Space:
+      stream.advance(1);
+      if (isAppropriateEndTagToken()) {
+        return beforeAttributeName;
+      }
+      return anythingElse;
+    case Char.Solidus:
+      stream.advance(1);
+      if (isAppropriateEndTagToken()) {
+        return selfClosingStartTag;
+      }
+      return anythingElse;
+    case Char.GreaterThanSign:
+      stream.advance(1);
+      if (isAppropriateEndTagToken()) {
+        emit(state.tag!);
+        return data;
+      }
+      return anythingElse;
+    default:
+      if (isUpperCase(char)) {
+        stream.advance(1);
+        if (state.tag !== null) {
+          state.tag.name += fromCharCode(char).toLowerCase();
+        } else {
+          state.tag = {
+            type: TokenType.EndTag,
+            name: fromCharCode(char).toLowerCase()
+          };
+        }
+        state.temporaryBuffer += fromCharCode(char);
+        break;
+      }
+      if (isLowerCase(char)) {
+        stream.advance(1);
+        if (state.tag !== null) {
+          state.tag.name += fromCharCode(char);
+        } else {
+          state.tag = {
+            type: TokenType.EndTag,
+            name: fromCharCode(char)
+          };
+        }
+        state.temporaryBuffer += fromCharCode(char);
+        break;
+      }
+      return anythingElse;
+  }
+};
+
+/**
+ * 8.2.4.26
+ * @see https://www.w3.org/TR/html/syntax.html#tokenizer-script-data-double-escape-start-state
+ */
+const scriptDataDoubleEscapeStart: Pattern = (stream, emit, state) => {
+  const char = stream.peek(0);
+  if (char === null) {
+    return scriptDataEscaped;
+  }
+  switch (char) {
+    case Char.CharacterTabulation:
+    case Char.LineFeed:
+    case Char.FormFeed:
+    case Char.Space:
+    case Char.Solidus:
+    case Char.GreaterThanSign:
+      stream.advance(1);
+      emit({ type: TokenType.Character, data: fromCharCode(char) });
+      if (state.temporaryBuffer === "script") {
+        return scriptDataDoubleEscaped;
+      }
+      return scriptDataEscaped;
+    default:
+      if (isUpperCase(char)) {
+        stream.advance(1);
+        state.temporaryBuffer += fromCharCode(char).toLowerCase();
+        emit({ type: TokenType.Character, data: fromCharCode(char) });
+        break;
+      }
+      if (isLowerCase(char)) {
+        stream.advance(1);
+        state.temporaryBuffer += fromCharCode(char);
+        emit({ type: TokenType.Character, data: fromCharCode(char) });
+        break;
+      }
```

Comment:
Same as previously; just check for an alphabetic `char` and lowercase indiscriminately.